cli/config/memorystore/store_test.go

```diff
@@ -0,0 +1,127 @@
+package memorystore
+
+import (
+	"testing"
+
+	"github.com/docker/cli/cli/config/types"
+	"gotest.tools/v3/assert"
+)
+
+func TestMemoryStore(t *testing.T) {
+	config := map[string]types.AuthConfig{
+		"https://example.com": {
+			Username:      "something-something",
+			ServerAddress: "https://example.com",
+			Auth:          "super_secret_token",
+		},
+	}
+
+	fallbackConfig := map[string]types.AuthConfig{
+		"https://only-in-file.com": {
+			Username:      "something-something",
+			ServerAddress: "https://only-in-file.com",
+			Auth:          "super_secret_token",
+		},
+	}
+
+	fallbackStore := New(WithAuthConfig(fallbackConfig))
+
+	memoryStore := New(WithAuthConfig(config), WithFallbackStore(fallbackStore))
+
+	t.Run("case=get credentials from memory store", func(t *testing.T) {
+		c, err := memoryStore.Get("https://example.com")
+		assert.NilError(t, err)
+		assert.Equal(t, c, config["https://example.com"])
+	})
+
+	t.Run("case=get credentials from fallback store", func(t *testing.T) {
+		c, err := memoryStore.Get("https://only-in-file.com")
```

Comment:
Even though nothing in this test _should_ be making networking connections, it's always better to use the designated domains for this ([rfc6761](https://www.rfc-editor.org/rfc/rfc6761.html). [rfc2606](https://www.rfc-editor.org/rfc/rfc2606). and related), e.g.;


```suggestion
		c, err := memoryStore.Get("https://only-in-file.example.com")
```

Or

```suggestion
		c, err := memoryStore.Get("https://only-in-file.test")
```



(Same for other domains we use in these tests)